### DISCUSSION TITLE
Is there any way to pass custom user profile properties as tokens of current login user inside Handle bar template ?

### DISCUSSION THREAD

**Discussion by MurugeshSandari** [microsoft-search/pnp-modern-search#1362] [Q&A]
Is there any way to pass custom user profile properties as tokens inside Handle bar template ?

**Comment by Laul0** [comment]
Hi @MurugeshSandari,

I am not sure to understand your point... You wish to display a custom People property, isn't it?

Anyway, it is possible to add a custom property to the user profiles and add this property in the Tenant Search Schema to retrieve it, refine results with it...
Once the property is referenced in the search schema, you can configure a slot to display it or manage the render of your Handlebars.

https://docs.microsoft.com/en-us/sharepoint/manage-search-schema

Hope this answer will help you

**Comment by Laul0** [comment]
After read again your question, I think my first answer does not match your needs 🧐
You want to use some properties of the current user profile to filter content or something like that, isn't it?
If this is your need, the SharePoint Search support several tokens including for the user profile: https://docs.microsoft.com/en-us/sharepoint/technical-reference/query-variables

Hope this answer will help you

**Comment by MurugeshSandari** [comment]
Hi @Laul0 

Thanks for your response. I can use this {{User.<PropertyName>}} in Query Template, to filter search results based on current login user.

where as my requirement is bit different,
I have to pass Current Login User Profile Properties to Handle Bars template and from there to Search Extensibility Web Component.

**Comment by wobba** [comment]
In v4 you can use `{{context.user.<something>}}`. If you flip to the debug template you can  see everything you can reference in a template. See https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/#context-tokens. We don't have support to load the props from a users profile.
